src/components/Patient/PatientRegistration.tsx

```diff
@@ -0,0 +1,1076 @@
+      pathParams: { id: form.state?.toString() || "" },
+    }),
+  });
+
+  const localBodyQuery = useQuery({
+    queryKey: ["localbodies", form.district],
+    enabled: !!form.district,
+    queryFn: query(routes.getLocalbodyByDistrict, {
+      pathParams: { id: form.district?.toString() || "" },
+    }),
+  });
+
+  const wardsQuery = useQuery({
+    queryKey: ["wards", form.local_body],
+    enabled: !!form.local_body,
+    queryFn: query(routes.getWardByLocalBody, {
+      pathParams: { id: form.local_body?.toString() || "" },
+    }),
+  });
+
+  const handlePincodeChange = async (value: string) => {
+    if (!validatePincode(value)) return;
+    if (form.state && form.district) return;
+
+    const pincodeDetails = await getPincodeDetails(
+      value,
+      careConfig.govDataApiKey,
+    );
+    if (!pincodeDetails) return;
+
+    const matchedState = statesQuery.data?.results?.find((state) => {
+      return includesIgnoreCase(state.name, pincodeDetails.statename);
+    });
+    if (!matchedState) return;
+    setForm((f) => ({
+      ...f,
+      state: matchedState.id,
+    }));
+    await new Promise((resolve) => setTimeout(resolve, 500));
+    const districts = await districtsQuery.refetch();
+
+    const matchedDistrict = districts.data?.find((district) => {
+      return includesIgnoreCase(district.name, pincodeDetails.districtname);
+    });
+    if (!matchedDistrict) return;
+    setForm((f) => ({
+      ...f,
+      district: matchedDistrict.id,
+    }));
+
+    setShowAutoFilledPincode(true);
+    setTimeout(() => {
+      setShowAutoFilledPincode(false);
+    }, 2000);
+  };
+
+  useEffect(() => {
+    const timeout = setTimeout(
+      () => handlePincodeChange(form.pincode?.toString() || ""),
+      1000,
+    );
+    return () => clearTimeout(timeout);
+  }, [form.pincode]);
+
+  const title = !patientId
+    ? t("add_details_of_patient")
+    : t("update_patient_details");
+
+  const errors = { ...feErrors, ...createPatientMutation.error };
+
+  const fieldProps = (field: keyof typeof form) => ({
+    value: form[field] as string,
+    onChange: (e: React.ChangeEvent<HTMLInputElement>) =>
+      setForm((f) => ({
+        ...f,
+        [field]: e.target.value === "" ? undefined : e.target.value,
+      })),
+  });
+
+  const selectProps = (field: keyof typeof form) => ({
+    value: (form[field] as string)?.toString(),
+    onValueChange: (value: string) =>
+      setForm((f) => ({ ...f, [field]: value })),
+  });
+
+  const handleDialogClose = (action: string) => {
+    if (action === "transfer") {
+      setShowTransferDialog(true);
+    } else if (action === "back") {
+      setShowTransferDialog(false);
+    } else {
+      setSuppressDuplicateWarning(true);
+      setShowTransferDialog(false);
+    }
+  };
+
+  const handleFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const errors: Record<string, string[]> = {};
+    const requiredFields: Array<keyof typeof form> = [
```

Comment:
These should move to a patient.ts file where the PatientModel and the patientCreateValidator should exist